### PR TITLE
Hts 548 adding not valid nino

### PR DESCRIPTION
**Pull request by Jackie1988**

### CODE REVIEW

File: test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala [hunk 1]
```diff
@@ -111,7 +111,7 @@ class HelpToSaveConnectorSpec extends TestSupport with GeneratorDrivenPropertyCh
           forAll(ineligibleResponseGen){ response ⇒
             val reason = Ineligible(response)
 
-            mockHttpGet(eligibilityURL)(Some(HttpResponse(200, responseJson = Some(Json.toJson(response)))))
+            mockHttpGet(eligibilityURL)(Some(HttpResponse(200, responseJson = Some(Json.toJson(ECResponseHolder(Some(response)))))))
```
Comment (AndyHWChung) [inline test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala]: Seem to be missing a unit test for the case when `None` is returned by the backend

File: app/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnector.scala [hunk 2]
```diff
@@ -121,20 +121,49 @@ class HelpToSaveConnectorImpl @Inject() (http: WSHttp)(implicit ec: ExecutionCon
...
   private val eligibilityURL = s"$helpToSaveUrl/help-to-save/eligibility-check"
 
-  // scalastyle:off magic.number
-  private def toEligibilityCheckResult(response: EligibilityCheckResponse): Either[String, EligibilityCheckResult] = {
-    response.resultCode match {
-      case 1     ⇒ Right(EligibilityCheckResult.Eligible(response))
-      case 2     ⇒ Right(EligibilityCheckResult.Ineligible(response))
-      case 3     ⇒ Right(EligibilityCheckResult.AlreadyHasAccount(response))
-      case other ⇒ Left(s"Could not parse eligibility result code '$other'. Response was '$response'")
+  private val emptyECResponse = EligibilityCheckResponse("There was no Eligibility Check Response", 2, "", -1)
```
[the remark this answers is not in the document]
Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnector.scala]: Updated.

File: test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala (hunk 1 above)
Comment (Jackie1988) [reply test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala]: Done

File: test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala [hunk 3]
```diff
@@ -124,22 +124,31 @@ class HelpToSaveConnectorSpec extends TestSupport with GeneratorDrivenPropertyCh
...
           val response = EligibilityCheckResponse("HtS account already exists", 3, reasonString, 1)
 
-          mockHttpGet(eligibilityURL)(Some(HttpResponse(200, responseJson = Some(Json.toJson(response)))))
+          mockHttpGet(eligibilityURL)(Some(HttpResponse(200, responseJson = Some(Json.toJson(ECResponseHolder(Some(response)))))))
 
           val result = connector.getEligibility()
           await(result.value) shouldBe Right(AlreadyHasAccount(response))
         }
 
+      "return a None when the given nino was not found to be in receipt of tax credit" in {
```
Comment (AndyHWChung) [inline test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala]: it's not really returning `None`, it's returning `Ineligible`

Comment (Jackie1988) [reply test/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnectorSpec.scala]: done

**Comment by AndyHWChung** [comment]
LGTM
